sktime/utils/validation/__init__.py

```diff
@@ -142,6 +142,8 @@ def check_n_jobs(n_jobs: int) -> int:
         return 1
     elif not is_int(n_jobs):
         raise ValueError(f"`n_jobs` must be None or an integer, but found: {n_jobs}")
+    elif n_jobs == -1:
```

Comment:
FYI @yarnabrina, due to recent discussion on `check_n_jobs`. Do you have a link to the issue I am trying to find? Search for `check_n_jobs` in the issue tracker does not find anything.

Reply:
@fkiraly @yarnabrina it brings another error:
```
AttributeError  
      1 cif = CanonicalIntervalForest(n_estimators=50, att_subsample_size=8, random_state=47)
----> 2 cif.fit(X_train,y_train)
      4 cif_preds = cif.predict(X_test)
      5 print("CIF Accuracy: " + str(metrics.accuracy_score(y_test, cif_preds)))

File /opt/hostedtoolcache/Python/3.9.18/x64/lib/python3.9/site-packages/sktime/classification/base.py:240, in BaseClassifier.fit(self, X, y)
    235         raise AttributeError(
    236             "self.n_jobs must be set if capability:multithreading is True"
    237         )
    239 # pass coerced and checked data to inner _fit
--> 240 self._fit(X,y)
    241 self.fit_time_ = int(round(time.time() * 1000)) - start
    243 # this should happen last: fitted state is set to True
```

Reply:
However, it is due to implementation of `Catch22Classifier`, so I don't know if it should be managed in this PR?

Reply:
In general, no dependent code should be broken by any pull request, without sufficient forewarning (see depercation policy: https://www.sktime.net/en/stable/developer_guide/deprecation.html)

I think there's three options:

* we make sure `Catch22` behaves identically for all input combinations pre and post this PR
* the reworked estimator is added as a third catch-22 estimator
* the reworked estimator is added as a third catch-22 estimator, and we run a single deprecation/change cycle to switch names in some sensible way

My preference would be option 1 if possible, since the PR claims to be a refactor plus extension of functionality, so if this is the case, current calls should not be impacted.

Reply:
if the change in `check_n_jobs` would change/break dependent code, we should not make it in this PR, but dedicate change/deprecation ot it. If @yarnabrina can help with finding the other discussion we had on this, the gist of it is that `check_n_jobs` occurs only in places where imo it is not necessary, and it could be refactored away, in favour of using `utils.parallel`.

Reply:
Maybe you are looking for #5990 ?

Reply:
yes, exactly.